src/main/java/net/sf/javaanpr/intelligence/Parser.java

```diff
@@ -293,4 +294,22 @@ public void addChar(char chr) {
         }
     }
 
+    public enum SyntaxAnalysisMode {
+        DO_NOT_PARSE,
+        ONLY_EQUAL_LENGTH,
+        EQUAL_OR_SHORTER_LENGTH
+    }
+
+    public static SyntaxAnalysisMode getSyntaxAnalysisModeFromInt(int syntaxAnalysisModeInt) {
```

Comment:
Note: once this function is in the enum, we could implement it as:
```
if (syntaxAnalysisModeInt < 0 || syntaxAnalysisModeInt >= values().length) {
    throw new IllegalArgumentException("Expected: 0, 1, or 2. Got: " + syntaxAnalysisModeInt);
} else {
    return values()[syntaxAnalysisModeInt];
}
```

However, I like you implementation better, as it might be easier to read in the long run, so I suggest we keep it as it is.